lib/Parse/ParseDecl.cpp

```diff
@@ -3945,7 +3945,10 @@ VarDecl *Parser::parseDeclVarGetSet(Pattern *pattern,
   if (!TyLoc.hasLocation()) {
     if (accessors.Get || accessors.Set || accessors.Addressor ||
         accessors.MutableAddressor) {
-      diagnose(pattern->getLoc(), diag::computed_property_missing_type);
+      SourceLoc locAfterPattern = pattern->getLoc().getAdvancedLoc(
```

Comment:
@DougGregor Would calling `getTypeOfExpressionWithoutApplying` here in limited situations (say, if we have a getter that looks like a single-expression closure) be the right way to fill this hole in?